python/pyspark/sql/dataframe.py

```diff
@@ -1830,6 +1830,43 @@ def agg(self, *exprs):
         """
         return self.groupBy().agg(*exprs)
 
+    @since(3.3)
+    def observe(self, observation, *exprs):
+        """Observe (named) metrics through an :class:`Observation` instance.
+
+        This method does not support streaming datasets.
+
+        .. versionadded:: 3.3.0
+
+        A user can retrieve the metrics by accessing `Observation.get`.
+
+        Parameters
+        ----------
+        observation : :class:`Observation`
+            an :class:`Observation` instance to obtain the metric.
+        exprs : list of :class:`Column`
+            column expressions (:class:`Column`).
+
+        Returns
+        -------
+        :class:`DataFrame`
+            the observed :class:`DataFrame`.
+
+        Examples
+        --------
+        >>> from pyspark.sql.functions import col, count, lit, max
+        >>> from pyspark.sql.observation import Observation
```

Comment:
Can you add the import of `Observation` to here: https://github.com/apache/spark/blob/master/python/pyspark/sql/__init__.py ?

So users can directly import it from `from pyspark.sql import Observation`. Let's fix the examples here as well

Reply:
I like that, done